broomcupboard.py

```diff
@@ -121,37 +135,44 @@ def get_equivalent_grades(grades):
      return equiv_grades
 
 def append_equivalent_grades(grades):
-     """
-     This function takes a list of climbing grades in either V-grade or font-grade format,
-     and appends equivalent grades in the other format to the original list in-place.
-     it does NOT accept range grades - these must be converted first
+    """
+    This function takes a list of climbing grades in either V-grade or font-grade format,
+    and appends equivalent grades in the other format to the original list in-place.
+    it does NOT accept range grades - these must be converted first
 
-     Parameters:
-     grades (list): A list of strings representing climbing grades. 
-     Each grade should be in either V-grade or font-grade format.
+    Parameters:
+    grades (list): A list of strings representing climbing grades. 
+    Each grade should be in either V-grade or font-grade format.
 
-     Returns:
-     int: 0 if the function executes successfully, otherwise an error code.
+    Returns:
+    int: 0 if the function executes successfully, otherwise an error code.
 
-     Raises:
-     TypeError: If the input grades is not a list.
-     ValueError: If any of the grades are not in either V-grade or font format.
-     Note:
-     The function uses regular expressions to match the grade format.
-     It then adds equivalent grades to the original list based on the matching format.
-     A deep copy of the input list is iterated over.
-     """
-     if not isinstance(grades, list):
-          raise TypeError(f'Expected list of strings of grades, got {type(grades)}')
-     gs = copy.deepcopy(grades)
-     for g in gs: # ensure don't end up in endless loop of adding grades to original list
-          if re.match(Vgrades_regex_pattern, g):
-               grades += V_to_font[g]               
-          elif re.match(font_grades_regex_pattern, g):
-               grades += font_to_V[g]      
-          else:
-               raise ValueError(f'Invalid grade format: {g}')         
-     return 0
+    Raises:
+    TypeError: If the input grades is not a list.
+    ValueError: If any of the grades are not in either V-grade or font format.
+    Note:
+    The function uses regular expressions to match the grade format.
+    It then adds equivalent grades to the original list based on the matching format.
+    A deep copy of the input list is iterated over.
+    It checks grades aren't already in the list before adding them.
+    """
+    if not isinstance(grades, list):
+        raise TypeError(f'Expected list of strings of grades, got {type(grades)}')
+    gs = set(copy.deepcopy(grades))
+    for g in gs: # ensure don't end up in endless loop of adding grades to original list
+        if re.match(Vgrades_regex_pattern, g):
+            newgrades = V_to_font[g]
+            for ng in newgrades:

```

Comment:
now checks if grade already in list before appending. prevents repetition if execute function twice